### PR TITLE
Added new charts and keptn example

### PR DESCRIPTION
**Pull request by nitishm**
- Added keptn-addons chart to orkestra. Addons include prometheus
  service and job executor services for keptn
- Created a new bookinfo example with istio (WIP)

TODO:
- Add keptn resources to bookinfo info example with istio

Signed-off-by: Nitish Malhotra <nitish.malhotra@gmail.com>

### CODE REVIEW

**Comment by nitishm** [comment]
Screen Shot 2021-09-30 at 4 36 51 PM

**Comment by nitishm** [comment]
Still needs a README - will add shortly

**Comment by nitishm** [comment]
@jonathan-innis PTAL

File: README.md [hunk 1]
```diff
@@ -127,12 +127,28 @@ The default executor is responsible for deploying the HelmRelease object passed
...
 Source code for the HelmRelease executor is available [here](https://github.com/Azure/helmrelease-workflow-executor)
 
-### Keptn Executor (Work in progress)
+### Keptn Executor
 
 The Keptn executor is an evaluation executor responsible for running tests on the deployed helm release using the Keptn API and Keptn evaluations engine. The Keptn executor is a custom executor that is chained to the default HelmRelease executor. This allows each release to be evaluated against a set of SLOs/SLIs before it is deployed/updated.
 
 Source code for the Keptn executor is available [here](https://github.com/Azure/keptn-workflow-executor)
 
+#### Argo workflow dashboard
```
Comment (jonathan-innis) [inline README.md:136]: Since the README is starting to get pretty long, I think it would be useful to have TOC at the top of the README

File: examples/keptn/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+The ConfigMap is used to configure the Keptn executor and contains the following:
+
+- *keptn-config.yaml* - This file contains the Keptn configuration.
+- *sli.yaml* - This file contains the SLI configuration.
+- *slo.yaml* - This file contains the SLO configuration.
+- *config.yaml* - This file contains the configuration for the `hey` load generator.
+
+```shell
+kubectl create -f bookinfo.yaml -n orkestra
```
Comment (jonathan-innis) [inline examples/keptn/README.md:61]: Specifying this command twice is unnecessary. Also, since the commands need to be executed together, can you make this into a single command with `\`?

File: examples/keptn/bookinfo-with-faults.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          name: istiod
+          version: 1.12.0-alpha.1
+        release:
+          targetNamespace: istio-system
+          timeout: 10m
+          values:
+            service:
+              type: ClusterIP
+    - name: istio-ingressgateway
+      dependencies: [istiod]
```
Comment (jonathan-innis) [inline examples/keptn/bookinfo-with-faults.yaml:40]: Maybe personal preference, but I feel like yaml is more readable when arrays are done like this

```yaml
dependencies:
   - istiod
```

File: pkg/graph/graph.go [hunk 4]
```diff
@@ -225,7 +226,7 @@ func (g *Graph) Reverse() *Graph {
 						node.Dependencies = append(node.Dependencies, executor.Name)
 					}
 				}
-				subChartNode.Executors[executor.Name].Executor = subChartNode.Executors[executor.Name].Executor.Reverse()
+				subChartNode.Executors[executor.Name].Executor = subTask.Executors[executor.Name].Executor.Reverse()
```
Comment (jonathan-innis) [inline pkg/graph/graph.go:229]: Why is this changed to subtask?

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nitishm) [reply README.md:136]: Sure I can add that

File: pkg/graph/graph.go (hunk 4 above)
Comment (nitishm) [reply pkg/graph/graph.go:229]: Oh shoot I should correct that - I was experimenting

File: README.md [hunk 5]
```diff
@@ -17,7 +15,35 @@ Orkestra is built on top of popular [CNCF](https://cncf.io/) tools and technolog
 - [Chartmuseum](https://chartmuseum.com/)
 - [Keptn](https://keptn.sh)
 
-<p align="center"><img src="./docs/assets/orkestra-core.png" width="750x" /></p>
+![Orkestra Core](docs/assets/orkestra-core.png)
```
Comment (jonathan-innis) [inline README.md:18]: Can we move the TOC to the very top?

File: examples/keptn/README.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+```shell
+helm upgrade --install orkestra chart/orkestra -n orkestra --create-namespace --set=keptn.enabled=true --set=keptn-addons.enabled=true  
+```
+
+### Scenario 1 : Successful Reconciliation
+
+The *bookinfo* application is deployed using the following Kubernetes manifests:
+
+```shell
```
Comment (jonathan-innis) [inline examples/keptn/README.md:48]: This command is still here twice. Any reason?

File: examples/keptn/bookinfo.yaml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            ambassador:
+              enabled: false
+        workflow:
+          - name: helmrelease
+            dependencies: []
+            type: helmrelease
+            params: nil
+          - name: keptn
+            dependencies: 
+            - "helmrelease"
```
Comment (jonathan-innis) [inline examples/keptn/bookinfo.yaml:76]: No need for quotes

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

File: examples/keptn/README.md (hunk 6 above)
Comment (nitishm) [reply examples/keptn/README.md:48]: redundant - let me remove it

File: README.md [hunk 8]
```diff
@@ -17,7 +15,35 @@ Orkestra is built on top of popular [CNCF](https://cncf.io/) tools and technolog
...
+
+## Table of Contents
+
+- [Orkestra](#orkestra)
+  - [Background and Motivation](#background-and-motivation)
+    - [Dependency Management in Helm](#dependency-management-in-helm)
+  - [What is Orkestra?](#what-is-orkestra)
```
Comment (jonathan-innis) [inline README.md:25]: Can we rename this to `Overview`?

File: README.md [hunk 9]
```diff
@@ -17,7 +15,35 @@ Orkestra is built on top of popular [CNCF](https://cncf.io/) tools and technolog
 - [Chartmuseum](https://chartmuseum.com/)
 - [Keptn](https://keptn.sh)
 
-<p align="center"><img src="./docs/assets/orkestra-core.png" width="750x" /></p>
+![Orkestra Core](docs/assets/orkestra-core.png)
+
+## Table of Contents
+
+- [Orkestra](#orkestra)
+  - [Background and Motivation](#background-and-motivation)
```
Comment (jonathan-innis) [inline README.md:23]: Can we switch the `Background and Motivation` sections with the `What is Orkestra?` section? We could also consider putting it under `Overview` if we grouped these all together
